### DISCUSSION TITLE
Can't publish to production from editor

### DISCUSSION THREAD

**Discussion by FeMaffezzolli** [swellstores/community#109] [Help]
Hey guys, I've been facing a problem regarding updating the prod environment. (dev env is getting updated normally though)

I noticed an error message on the app.

![image](https://user-images.githubusercontent.com/26548010/214361271-f068b5c9-ffb4-4f44-8c55-bf22ce2cbc38.png)

What should I do?

**Comment by Strongeyed** [comment]
I've been experiencing issues with publishing my stores for a couple of weeks now (3) it's actually present in two individual stores that I run separately from each other through Swell. Did you have any luck solving this issue, keen to get some tips!

![Screenshot 2023-03-17 at 11 17 39 AM](https://user-images.githubusercontent.com/128098520/225764943-497c066a-ba62-4b8d-8e65-b265d50e715e.png)

**Comment by rodrigoddalmeida** [comment]
@FeMaffezzolli @Robin-DG could you update your storefronts (Origin) to the latest version, then publish? That should take care of the error you see. You can do it through the Admin Dashboard UI or by rebasing/merging from Origin's public repository if you have custom code.

**Reply by Strongeyed** [reply]
@rodrigoddalmeida I've updated to version V2.0.14 and was able to publish one, issues with the Slugs (how they appear) are now on the website and I can't now publish again.

Couple of examples attached, also reported this through the chat functionality on the website.

![Screenshot 2023-03-22 at 2 01 53 PM](https://user-images.githubusercontent.com/128098520/226775047-32b63e9f-6c4c-4408-85e5-f2e163cd73dc.png)
![Screenshot 2023-03-22 at 2 02 13 PM](https://user-images.githubusercontent.com/128098520/226775084-d2f9bd4f-cdf7-483e-8775-8f13d46e0e03.png)
![Screenshot 2023-03-22 at 2 02 24 PM](https://user-images.githubusercontent.com/128098520/226775112-efdda8b9-7bfb-470e-b088-77e6766e6440.png)

**Reply by rodrigoddalmeida** [reply]
@Robin-DG there's a fix coming up soon - I'll update back when it is released.

**Reply by rodrigoddalmeida** [reply]
@Robin-DG a fix should be released just now @ v2.0.15

**Reply by Strongeyed** [reply]
@rodrigoddalmeida The fix works perfectly and changes are shown on the dev end, however I can't publish them due to the issues described at the top of this thread. When we're publishing it goes from 'initialising, please wait' to changes are live.

**Reply by rodrigoddalmeida** [reply]
@Robin-DG got it. I'll take a look.

**Reply by Strongeyed** [reply]
@rodrigoddalmeida Thanks! Let me know if you need anything from my side.

**Comment by Strongeyed** [comment]
In the meanwhile, is there a way to publish the storefront in a different way? I’m getting a lot of customers reaching out by email and messenger that the website is broken so it’d be great if there’s a temporary fix.

**Reply by Strongeyed** [reply]
Anyone? Any suggestions?😅

**Reply by ericingram** [reply]
@Robin-DG we're pretty close to releasing a change to storefronts that will make it possible to run the theme editor with an external host like Vercel or Netlify. Have you tried deploying it there instead? (While @rodrigoddalmeida looks into this issue)

**Reply by Strongeyed** [reply]
@ericingram I'll try to set this up as it's been 4 weeks now without being able to publish any of my stores. Is there any documentation I could follow to do this? @rodrigoddalmeida Would still appreciate if we could fix this as I'm not sure if I'll be able to make Vercel work for us 👍 The weird thing is that I was able to publish the store (once) after I did the update (prior to the slugs being displayed) but currently we are back to square one where the publishing doesn't work.

https://user-images.githubusercontent.com/128098520/228350020-02f873b1-1a0c-43e1-9555-e199fed0db2e.mov

**Reply by Strongeyed** [reply]
Both my stores report back something similar in the Chrome console:

DevTools failed to load source map: Could not load content for https://xxxxxxx.store/admin/public/static/js/main.b3707eeb.js.map: Unexpected token '<', "<!doctype "... is not valid JSON

Continued by: 

main.b3707eeb.js:2 alias matches current distinct_id - sk

![Screenshot 2023-03-29 at 9 23 19 AM](https://user-images.githubusercontent.com/128098520/228358218-b81d74db-2d60-4070-9449-a7f73c9abace.png)
ipping api call.

After a while it reports:

main.b3707eeb.js:2          PUT https://lambda.swell.store/api/files/xxxxxxx--dev/git/commit 400 (Bad Request)
Error: [object Object]
With a whole string of references.

**Reply by rodrigoddalmeida** [reply]
@Robin-DG we're still looking for a fix for the publishing issue - I'll update as soon as it is released.

**Reply by Strongeyed** [reply]
@rodrigoddalmeida hope you had a great weekend! Any updates on this?

**Reply by ericingram** [reply]
That's great! Thanks for the update

**Reply by logeshswell** [reply]
@Robin-DG @FeMaffezzolli Can you share your store link to assist you better?

**Reply by FeMaffezzolli** [reply]
@logeshswell , thanks for your reply. Unfortunately I had to stop the project  I was working on so I removed the store. Thanks anyway.
